### PR TITLE
Cleanup some code

### PR DESCRIPTION
**Pull request by ricardoV94**
## Description

Just some trivial stuff like not doing instance checks in `node_rewriter`s that have `tracks` and not accessing the same attribute multiple times in a loop

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [x] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

**Review by Armavica — APPROVED** [review]
Nice! I allowed myself to add a commit to remove more of `not x.owner`, feel free to rebase or remove it you left them on purpose.

I also have two questions:
1. Do you think we also should rewrite the `x.owner` into `x.owner is not None`?
2. Some rewrites return `False` and some return `None`, should they all return `None` eventually?

**Comment by ricardoV94** [comment]
Thanks @Armavica that's awesome.

1. I thought about it, but I felt it hurts readability (even if it's slightly slower)
2. I don't remember `None` / `False`. I think there was some idea about distinguishing between rewrites that will never apply or may apply but not just now. We are not treating making use of this distinction AFAICT, so I would be happy to make everything `None` like you suggested.

File: pytensor/tensor/rewriting/subtensor.py [hunk 1]
```diff
@@ -1445,7 +1445,7 @@ def local_adv_sub1_adv_inc_sub1(fgraph, node):
     if not isinstance(node.op, AdvancedSubtensor1):
         return
     inp = node.inputs[0]
-    if not inp.owner or not isinstance(inp.owner.op, AdvancedIncSubtensor1):
+    if inp.owner is None or not isinstance(inp.owner.op, AdvancedIncSubtensor1):
```
Comment (ricardoV94) [inline pytensor/tensor/rewriting/subtensor.py:1448]: This pattern I like to switch into `if not (inp.owner and isinstance(inp.owner.op, AdvancedIncSubtensor1)`. It explicitly states you are looking for an input that's an AdvancedIncSubtensor, and getting out if that's not the case.

I tried to match some of these cases with regex (maybe only some files), in one of the commits.

File: pytensor/graph/basic.py [hunk 2]
```diff
@@ -1896,9 +1896,9 @@ def equal_computations(
             if isinstance(x, Constant):
                 return np.array_equal(x.data, y)
             return False
-        if x.owner and not y.owner:
+        if x.owner and y.owner is None:
             return False
-        if y.owner and not x.owner:
+        if y.owner and x.owner is None:
```
Comment (ricardoV94) [inline pytensor/graph/basic.py:1901]: In this case the if `x.owner is not None and ...` seems fine. Maybe also all, anyway see my reasoning in the main thread.

File: pytensor/tensor/rewriting/basic.py [hunk 3]
```diff
@@ -662,7 +662,7 @@ def local_cast_cast(fgraph, node):
         return
     x = node.inputs[0]
     if (
-        not x.owner
+        x.owner is None
         or not isinstance(x.owner.op, Elemwise)
         or not isinstance(x.owner.op.scalar_op, ps.Cast)
```
Comment (ricardoV94) [inline pytensor/tensor/rewriting/basic.py:667]: This is like that one, but no need to try to fish everything I guess

File: pytensor/graph/basic.py [hunk 4]
```diff
@@ -1896,11 +1896,10 @@ def equal_computations(
...
                 return np.array_equal(x.data, y)
             return False
-        if x.owner and not y.owner:
+        xy_have_owner = (x.owner is None, y.owner is None)
+        if any(xy_have_owner) and not all(xy_have_owner):
             return False
-        if y.owner and not x.owner:
-            return False
-        if x.owner and y.owner:
+        if all(xy_have_owner):
```
Comment (ricardoV94) [inline pytensor/graph/basic.py:1902]: I find this a bit less readable now

Comment (Armavica) [reply pytensor/graph/basic.py:1902]: Ah and I even reversed it :upside_down_face:

**Comment by ricardoV94** [comment]
Looks great @Armavica anything else or shall we merge?

**Comment by Armavica** [comment]
@ricardoV94 All good for me!

**Pull request merged**